### DISCUSSION TITLE
Delight: What if users to join a space in their first session?

### DISCUSSION THREAD

**Discussion by daniellekirkwood** [element-hq/element-meta#516] [Ideas]
HMW support a user wishing to join a space in their first session?

The Delight team are exploring options for this in Q2 and Q3 of 2022 as part of the larger New user Journey epic

**Comment by duxovni** [comment]
(Marking this as "Needs Info" for the Web team triage since I'm not clear on what the proposal is, but if this is mostly a note-to-self and you're already handling it, then no worries)

**Comment by daniellekirkwood** [comment]
> (Marking this as "Needs Info" for the Web team triage since I'm not clear on what the proposal is, but if this is mostly a note-to-self and you're already handling it, then no worries)

Thank you! Yes, I created this as an empty issue and will add more info asap. Still, let's leave the label on it!

**Comment by daniellekirkwood** [comment]
"should explore ways to give homeserver admins an easy way from GUI to setup auto-join rooms when registering...

But this means exposing some admin functionality from a standardised API (which means we need to go through an MSC \o/ )"

**Comment by t3chguy** [comment]
Given auto-join rooms is a Synapse-specific concept, it isn't something we should be doing. The Matrix Spec doesn't really have a concept of homeserver admins so adding features specifically for them becomes much harder as first you'd need to MSC that.

**Comment by germain-gg** [comment]
The foundations of an administration API exists as part of https://github.com/matrix-org/matrix-spec-proposals/pull/3593 but this MSC's scope is probably a bit too large for what we'd like to do.

Another interesting thing to note is that the auto room join config lives in synapses yaml config, which requires a synapse restart to be picked up.

The synapse admin management is not really standardised at the moment. We should be able to add a way to manage that in the EMS management console but this will not solve the problem holistically
Projects like https://github.com/Awesome-Technologies/synapse-admin would also need to be considered if we're looking to have a broader impact

A good way to validate our theories here, is to apply those changes on the large public homeserver by suggesting some configuration to admins and watch metrics that we can gather from there

**Comment by turt2live** [comment]
@daniellekirkwood can we get an update on this issue please? The description is still blank, and queued for closing.

**Comment by daniellekirkwood** [comment]
Sure. Why is it queued for closing?

**Comment by t3chguy** [comment]
@daniellekirkwood because it has been marked as Needs Info for a month
